docs/modules/ROOT/pages/api/access.adoc

```diff
@@ -0,0 +1,497 @@
+event.
+
+Requirements:
+
+- the caller must have ``role``'s admin role.
+
+[.contract-item]
+[[IAccessControl-revoke_role]]
+==== `[.contract-item-name]#++revoke_role++#++(role: felt252, account: ContractAddress)++` [.item-kind]#external#
+
+Revokes `role` from `account`.
+
+If `account` had been granted `role`, emits a {RoleRevoked} event.
+
+Requirements:
+
+- the caller must have ``role``'s admin role.
+
+[.contract-item]
+[[IAccessControl-renounce_role]]
+==== `[.contract-item-name]#++renounce_role++#++(role: felt252, account: ContractAddress)++` [.item-kind]#external#
+
+Revokes `role` from the calling account.
+
+Roles are often managed via {grant_role} and {revoke_role}. This function's
+purpose is to provide a mechanism for accounts to lose their privileges
+if they are compromised (such as when a trusted device is misplaced).
+
+If the calling account had been granted `role`, emits a {RoleRevoked}
+event.
+
+Requirements:
+
+- the caller must be `account`.
+
+[#IAccessControl-Events]
+==== Events
+
+[.contract-item]
+[[IAccessControl-RoleAdminChanged]]
+==== `[.contract-item-name]#++RoleAdminChanged++#++(role: felt252, previous_admin_role: ContractAddress, new_admin_role: ContractAddress)++` [.item-kind]#event#
+
+Emitted when `new_admin_role` is set as ``role``'s admin role, replacing `previous_admin_role`
+
+`DEFAULT_ADMIN_ROLE` is the starting admin for all roles, despite
+{RoleAdminChanged} not being emitted signaling this.
+
+[.contract-item]
+[[IAccessControl-RoleGranted]]
+==== `[.contract-item-name]#++RoleGranted++#++(role: felt252, account: ContractAddress, sender: ContractAddress)++` [.item-kind]#event#
+
+Emitted when `account` is granted `role`.
+
+`sender` is the account that originated the contract call, an admin role
+bearer.
+
+[.contract-item]
+[[IAccessControl-RoleRevoked]]
+==== `[.contract-item-name]#++RoleRevoked++#++(role: felt252, account: ContractAddress, sender: ContractAddress)++` [.item-kind]#event#
+
+Emitted when `account` is revoked `role`.
+
+`sender` is the account that originated the contract call:
+
+- if using `revoke_role`, it is the admin role bearer.
+- if using `renounce_role`, it is the role bearer (i.e. `account`).
+
+[.contract]
+[[AccessControl]]
+=== `++AccessControl++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/access/accesscontrol/accesscontrol.cairo[{github-icon},role=heading-link]
+
+:assert_only_role: xref:#AccessControl-assert_only_role
+:grant_role: xref:#AccessControl-grant_role[grant_role]
+:revoke_role: xref:#AccessControl-revoke_role[revoke_role]
+
+```javascript
+use openzeppelin::access::accesscontrol::AccessControl;
+```
+
+Contract module that allows children to implement role-based access control mechanisms.
+Roles are referred to by their `felt252` identifier:
+
+```javascript
+// sn_keccak('MY_ROLE')
+const MY_ROLE: felt252 =
+    0x3c877e40edb41710f0baf588c878ee15a04499b06ae8c98cf488875d91a7213;
+```
```

Comment:
```suggestion
```javascript
// 0x3c877e40edb41710f0baf588c878ee15a04499b06ae8c98cf488875d91a7213
const MY_ROLE: felt252 = sn_keccak('MY_ROLE');
```
shouldn't it be like this?

Reply:
`sn_keccak` is not actually a function in Cairo, but I updated it to use the `selector!` macro.

Reply:
then what we gain by putting `sn_keccak` in there? to me it's confusing rather than clarifying

Reply:
To me, it shows that the selector macro is returning the sn_keccak. If you think that doesn't add value let me know and will remove it.

Reply:
i do think it's a bit confusing, especially if there's no `sn_keccak` function available, or is it available somewhere else? if we just want to say it's keccak we can simply say that. but I also don't think this is the place to learn how selectors are computed, we just want to compute it. no strong opinion on leaving a comment.

Reply:
Removed!